public/js/components/AtomEdit/CustomEditors/AudioFields/AutomaticDataFetch.js

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react';
+import {ManualDataInput} from "./ManualDataInput";
+
+class AutomaticDataFetch extends React.Component {
+
+  static propTypes = {
+    getAudioPageData: PropTypes.func,
+    onUpdate: PropTypes.func.isRequired,
+    onFormErrorsUpdate: PropTypes.func,
+    error: PropTypes.string,
+    audioPageData: PropTypes.object,
+    atom: atomPropType.isRequired,
+  };
+
+  state = {
+    audioPageUrl: '',
+    urlError: ''
+  };
+
+  getAudioDataFromCapi () {
+    if(this.checkUrlIsAudioPage(this.state.audioPageUrl)){
+      this.props.getAudioPageData(this.state.audioPageUrl, this.props.atom);
+      this.setState({urlError: ''})
+    } else {
+      this.setState({urlError: 'The url is not a valid Guardian audio page'})
+    }
+  }
+
+  checkUrlIsAudioPage (url) {
+    return url.startsWith('https://www.theguardian.com/news/audio');
+  }
+
+  handleChange (e) {
+    this.setState({audioPageUrl: e.target.value});
+  }
+
+  render () {
+    return (
+      <div>
+        <div className="audio--automatic-selection-box">
+          <h3>Get audio data automatically</h3>
+          <label>Paste the url of a Guardian audio page</label>
+          <input id="audio-url" className="form__field" type="text" value={this.state.audioPageUrl} onChange={this.handleChange.bind(this)}></input>
+          <button type="button" className="form__btn-heading__btn" onClick={this.getAudioDataFromCapi.bind(this)}>Get values</button>
+          <p>{ this.state.urlError }</p>
+          <p>{ this.props.audioPageData ? this.props.audioPageData.message : "" }</p>
+
+        </div>
+
+        <div className="audio--manual-selection-box">
+          <h3>Or manually add the data in here</h3>
+          <ManualDataInput atom={this.props.atom} onUpdate={this.props.onUpdate} audioPageData={this.props.audioPageData}/>
+        </div>
+      </div>
+    )
+  }
+}
+
+//REDUX CONNECTIONS
+import { connect } from 'react-redux';
```

Comment:
I'm surprised import statements are allowed in the middle of a module 🤔

Reply:
well, it is the convention in this repo. But am happy to move them up to the top, for this repo at least

Reply:
No no, I was just surprised is all 😅